tests/foreman/ui_airgun/test_hardwaremodel.py

```diff
@@ -0,0 +1,78 @@
+"""Test class for Hardware Model UI
+
+:Requirement: HardwareModel
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: UI
+
+:TestType: Functional
+
+:CaseImportance: Low
+
+:Upstream: No
+"""
+from fauxfactory import gen_string
+from nailgun import entities
+from pytest import raises
+
+from robottelo.decorators import tier2, upgrade
+from tests.foreman.ui_airgun.test_host import create_fake_host
+
+
+@tier2
+@upgrade
+def test_positive_end_to_end(session, module_org, module_loc):
+    """Perform end to end testing for hardware model component
+
+    :id: 93663cc9-7c8f-4f43-8050-444be1313bed
+
+    :expectedresults: All expected CRUD actions finished successfully
+
+    :CaseLevel: Integration
+
+    :CaseImportance: High
+    """
+    name = gen_string('alpha')
+    model = gen_string('alphanumeric')
+    vendor_class = gen_string('alpha')
+    info = gen_string('alpha')
+    new_name = gen_string('alpha')
+    host_template = entities.Host(organization=module_org, location=module_loc)
+    host_template.create_missing()
+    with session:
+        # Create new hardware model
+        session.hardwaremodel.create({
+            'name': name,
+            'hardware_model': model,
+            'vendor_class': vendor_class,
+            'info': info,
+        })
+        assert session.hardwaremodel.search(name)[0]['Name'] == name
+        hm_values = session.hardwaremodel.read(name)
+        assert hm_values['name'] == name
+        assert hm_values['hardware_model'] == model
+        assert hm_values['vendor_class'] == vendor_class
+        assert hm_values['info'] == info
+        # Create host with associated hardware model
+        host_name = create_fake_host(
+            session,
+            host_template,
+            extra_values={'additional_information.hardware_model': name}
+        )
+        host_values = session.host.read(host_name)
+        assert host_values['additional_information']['hardware_model'] == name
+        # Update hardware model with new name
+        session.hardwaremodel.update(name, {'name': new_name})
```

Comment:
```python
assert not session.hardwaremodel.search(name)
```

Reply:
perhaps we should also check that 'additional_information.hardware_model' has also changed to new_name